### PR TITLE
feat: add ready endoint to loki-push-api server

### PR DESCRIPTION
**Pull request by Yasmine92**
**What this PR does / why we need it**:

When trying to expose `loki_push_api` service (in our case using GKE ingress), we need a health check target endpoint that would return a 200 response code.
this PR contains a simple addition for a HTTP handler that handles the `/ready` path, along with the necessary tests.

**Which issue(s) this PR fixes**:
https://github.com/grafana/loki/issues/8551

**Special notes for your reviewer**:

**Checklist**
- [x] Reviewed the [`CONTRIBUTING.md`](https://github.com/grafana/loki/blob/main/CONTRIBUTING.md) guide (**required**)
- [x] Documentation added
- [x] Tests updated
- [ ] `CHANGELOG.md` updated
- [ ] Changes that require user attention or interaction to upgrade are documented in `docs/sources/upgrading/_index.md`

### CODE REVIEW

File: clients/pkg/promtail/targets/lokipush/pushtarget_test.go [hunk 1]
```diff
@@ -200,3 +201,66 @@ func TestPlaintextPushTarget(t *testing.T) {
...
+	wantedResponse := "ready"
+	if responseBody != wantedResponse {
+		t.Errorf("response_body %q, want %q", responseBody, wantedResponse)
+	}
+	wantedCode := "200"
+	if responseCode != wantedCode {
+		t.Errorf("response_body %q, want %q", responseCode, wantedCode)
+	}
+
+	_ = pt.Stop()
```
Comment (chaudum) [inline clients/pkg/promtail/targets/lokipush/pushtarget_test.go:264]: We should put the stop function into a `t.Cleanup` function, which ensures the method is called even when tests fail.

```go
	pt, err := NewPushTarget(logger, eh, []*relabel.Config{}, "job3", config)
	require.NoError(t, err)
    t.Cleanup(func(){
        _ = pt.Stop()
    })
```

**Review by chaudum — COMMENTED** [review]
Otherwise LGTM

File: clients/pkg/promtail/targets/lokipush/pushtarget_test.go (hunk 1 above)
Comment (Yasmine92) [reply clients/pkg/promtail/targets/lokipush/pushtarget_test.go]: Thank you for reviewing my PR, I added the adjustment you suggested :)

File: docs/sources/clients/promtail/configuration.md [hunk 2]
```diff
@@ -877,6 +877,8 @@ Note the `server` configuration is the same as [server](#server).
 Promtail also exposes a second endpoint on `/promtail/api/v1/raw` which expects newline-delimited log lines.
 This can be used to send NDJSON or plaintext logs.
 
+The readiness of the loki_push_api server can be checked using the endpoint `/ready`
```
Comment (chaudum) [inline docs/sources/clients/promtail/configuration.md:880]:
```suggestion
The readiness of the loki_push_api server can be checked using the endpoint `/ready`.
```

**Review by chaudum — APPROVED** [review]
Thanks!

**Pull request merged**